scraper/src/mindtouch2zim/context.py

```diff
@@ -77,68 +81,56 @@ class Context:
     # URL to Mindtouch instance
     library_url: str
 
-    def __init__(self) -> None:
-        if path := os.getenv("MINDTOUCH_TMP"):
-            self.tmp_folder = Path(path)
+    # ZIM properties
+    creator: str
+    publisher: str = "openZIM"
+    file_name: str = "{name}_{period}"
+    name: str
+    title: str
+    description: str
+    long_description: str | None = None
+    tags: list[str] | None = None
+
+    # secondary color of the UI
+    secondary_color: str = "#FFFFFF"
+
+    # Content filters
+    page_title_include: re.Pattern | None = None
+    page_id_include: list[str] | None = None
+    page_title_exclude: re.Pattern | None = None
+    root_page_id: str | None = None
+
+    @classmethod
+    def setup(cls, **kwargs):
+        new_instance = cls(**kwargs)
+        if cls._instance:
```

Comment:
Can you elaborate on the use case for this? I see none. But I see how this can be accidentally called and be a problem.

Reply:
I opened https://github.com/openzim/mindtouch/issues/112 to still merge this and be able to continue the discussion